server/models/annotation.py

```diff
@@ -0,0 +1,534 @@
+            baseShapeSchema,
+            {
+                'type': 'object',
+                'properties': {
+                    'type': {
+                        'type': 'string',
+                        'enum': ['polyline']
+                    },
+                    'points': {
+                        'type': 'array',
+                        'items': coordSchema,
+                        'minItems': 2,
+                    },
+                    'fillColor': colorSchema,
+                    'closed': {
+                        'type': 'boolean',
+                        'description': 'polyline is open if closed flag is '
+                                       'not specified'
+                    },
+                },
+                'required': ['type', 'points'],
+                'patternProperties': baseShapePatternProperties,
+                'additionalProperties': False
+            }
+        ]
+    }
+
+    baseRectangleShapeSchema = {
+        'allOf': [
+            baseShapeSchema,
+            {
+                'type': 'object',
+                'properties': {
+                    'type': {'type': 'string'},
+                    'center': coordSchema,
+                    'width': {
+                        'type': 'number',
+                        'minimum': 0
+                    },
+                    'height': {
+                        'type': 'number',
+                        'minimum': 0
+                    },
+                    'rotation': {
+                        'type': 'number',
+                        'description': 'radians counterclockwise around normal',
+                    },
+                    'normal': coordSchema,
+                    'fillColor': colorSchema
+                },
+                'decription': 'normal is the positive z-axis unless otherwise '
+                              'specified',
+                'required': ['type', 'center', 'width', 'height', 'rotation'],
+                # 'patternProperties': baseShapePatternProperties,
+                'additionalProperties': True,
+            }
+        ]
+    }
+    baseRectangleShapePatternProperties = {
+        '^%s$' % properyName: {} for properyName in six.viewkeys(
+            baseRectangleShapeSchema['allOf'][1]['properties'])
+        if properyName != 'type'
+    }
+    baseRectangleShapePatternProperties.update(baseShapePatternProperties)
+    rectangleShapeSchema = {
+        'allOf': [
+            baseRectangleShapeSchema,
+            {
+                'type': 'object',
+                'properties': {
+                    'type': {
+                        'type': 'string',
+                        'enum': ['rectangle']
+                    },
+                },
+                'required': ['type'],
+                'patternProperties': baseRectangleShapePatternProperties,
+                'additionalProperties': False
+            }
+        ]
+    }
+    rectangleGridShapeSchema = {
```

Comment:
Charles has found it useful in SlideAtlas to have a grid (a regular array of rectangles) at a specific scale and orientation to add in scale analysis of slides.  Rather than just be the frame of a rectangle (as in rectangleShapeSchema), the grid adds `widthSubdivisions` and `heightSubdivisions`, and, as its name suggests, renders as a grid.